src/viadot/sources/epicor.py

```diff
@@ -0,0 +1,497 @@
+                        ship_value = shipto.find(ship_param).text
+                    except AttributeError:
+                        ship_value = None
+                    ship_parameter = {ship_param: ship_value}
+                    ship_dict.update(ship_parameter)
+                ship_address = ShipToAddress(**ship_dict)
+
+                for cust_param in Customer.__dict__.get("__annotations__"):
+                    try:
+                        cust_value = customer.find(cust_param).text
+                    except AttributeError:
+                        cust_value = None
+                    if cust_param == "ShipTos":
+                        cust_parameter = {cust_param: ship_address}
+                    else:
+                        cust_parameter = {cust_param: cust_value}
+                    customer_params_dict.update(cust_parameter)
+                cust_info = Customer(**customer_params_dict)
+                my_dict = cust_info.dict()
+                final_df = final_df.append(
+                    pd.json_normalize(my_dict, max_level=2), ignore_index=True
+                )
+    return final_df
+
+
+class EpicorCredentials(BaseModel):
+    host: str
+    port: int = 443
+    username: str
+    password: str
+
+
+class Epicor(Source):
+    def __init__(
+        self,
+        base_url: str,
+        filters_xml: str,
+        credentials: dict[str, Any] | None = None,
+        config_key: str | None = None,
+        validate_date_filter: bool = True,
+        start_date_field: str = "BegInvoiceDate",
+        end_date_field: str = "EndInvoiceDate",
+        *args,
+        **kwargs,
+    ):
+        """Class to connect to Epicor API and pasere XML output into a pandas DataFrame.
+
+        Args:
+            base_url (str, required): Base url to Epicor.
+            filters_xml (str, required): Filters in form of XML. The date filter
+                 is required.
+            credentials (dict[str, Any], optional): Credentials to connect with
+            Epicor API containing host, port, username and password.Defaults to None.
+            config_key (str, optional): Credential key to dictionary where details
+                 are stored.
+            validate_date_filter (bool, optional): Whether or not validate xml
+                date filters. Defaults to True.
+            start_date_field (str, optional) The name of filters field containing
+                start date.Defaults to "BegInvoiceDate".
+            end_date_field (str, optional) The name of filters field containing
+                end date. Defaults to "EndInvoiceDate".
+        """
+        raw_creds = credentials or get_source_credentials(config_key) or {}
+        validated_creds = EpicorCredentials(**raw_creds).dict(
+            by_alias=True
+        )  # validate the credentials
+
+        self.credentials = validated_creds
+        self.config_key = config_key
+        self.base_url = base_url
+        self.filters_xml = filters_xml
```

Comment:
I think this is per-request param, so should be a parameter in relevant methods (eg. `validate_filter()`, `get_xml_response()`)  rather than attribute.

It will also make the methods easier to unit test since they won't depend on state (eg. you could test `validate_filter()` by passing various test filters as a param, rather than creating separate `Epicor` instances for each test).